feature_engine/selection/information_value.py

```diff
@@ -0,0 +1,328 @@
+    """
+    InformationValue() calculates the information value (IV) for each variable.
+    The transformer is only compatible with categorical variables (type 'object'
+    or 'categorical') and binary classification.
+
+    You can pass a list of variables to score. Alternatively, the
+    transformer will find and score all categorical variables (type 'object'
+    or 'categorical').
+
+    IV will allow you to assess each variable's independent contribution to
+    the target variable and rank the variables in terms of their univariate
+    predictive strength.
+
+    Parameters
+    ----------
+    {variables}
+
+    sort_values: bool, default=False
+        Determines whether to sort the variables in ascending order after
+        the information values are calculated.
+
+    {errors}
+
+    {ignore_format}
+
+    Attributes
+    ----------
+    {variables_}
+
+    {feature_names_in}
+
+    {n_features_in}
+
+    class_diff_encoder_dict_:
+         The difference between the binomial distributions of positive and
+         negative classes for each unique value of the selected categorical
+         variables.
+
+    woe_encoder_dict_:
+        Contains the weights of evidence (WoE) for all the unique values of
+        the selected categorical variables.
+
+    information_values_:
+        A dictionary comprised of the information values for the selected
+        categorical variables.
+
+    Methods
+    -------
+    fit:
+        Calculates the information values for the selected categorical
+        variables.
+
+    {fit_transform}
+
+    transform:
+        Returns a dataframe with the variables and their corresponding
+        information values.
+
+    See Also
+    --------
+    feature_engine.encoding.WoEEncoder
+    """
+
+    def __init__(
+            self,
+            variables: Union[None, int, str, List[Union[str, int]]] = None,
+            sort_values: bool = False,
+            ignore_format: bool = False,
+            errors: str = "ignore",
+    ) -> None:
+
+        if not isinstance(sort_values, bool):
+            raise ValueError(
+                "sort_values must be a boolean variable. Got {sort_values} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.sort_values = sort_values
+
+        # parameters are checked when WoEEncoder is instatiated
+        self.ignore_format = ignore_format
+        self.errors = errors
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """
+        Learn the information value.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+            Can be the entire dataframe, not just the categorical variables.
+
+        y: pandas series of shape = [n_samples, ]
+            Target, must be binary.
+
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
```

Comment:
The class WoEEncoder and this class, both need to calculate the WoE.

So we need to extract that calculation into a function, that both WoE and this class can use to return pos and neg. This way, we do not repeat the code /logic